mllib/src/main/scala/org/apache/spark/mllib/clustering/LDA.scala

```diff
@@ -0,0 +1,472 @@
+   * ("theta").  We use a symmetric Dirichlet prior.
+   *
+   * This value should be > 0.0, where larger values mean more smoothing (more regularization).
+   * If set to -1, then topicSmoothing is set automatically.
+   *  (default = -1 = automatic)
+   *
+   * Automatic setting of parameter:
+   *  - For EM: default = (50 / k) + 1.
+   *     - The 50/k is common in LDA libraries.
+   *     - The +1 follows Asuncion et al. (2009), who recommend a +1 adjustment for EM.
+   */
+  def getTopicSmoothing: Double = topicSmoothing
+
+  def setTopicSmoothing(topicSmoothing: Double): this.type = {
+    require(topicSmoothing > 0.0 || topicSmoothing == -1.0,
+      s"LDA topicSmoothing must be > 0 (or -1 for auto), but was set to $topicSmoothing")
+    if (topicSmoothing > 0.0 && topicSmoothing <= 1.0) {
+      logWarning(s"LDA.topicSmoothing was set to $topicSmoothing, but for EM, we recommend > 1.0")
+    }
+    this.topicSmoothing = topicSmoothing
+    this
+  }
+
+  /**
+   * Term smoothing parameter (commonly named "eta").
+   *
+   * This is the parameter to the Dirichlet prior placed on the per-topic word distributions
+   * (which are called "beta" in the original LDA paper by Blei et al., but are called "phi" in many
+   *  later papers such as Asuncion et al., 2009.)
+   *
+   * This value should be > 0.0.
+   * If set to -1, then termSmoothing is set automatically.
+   *  (default = -1 = automatic)
+   *
+   * Automatic setting of parameter:
+   *  - For EM: default = 0.1 + 1.
+   *     - The 0.1 gives a small amount of smoothing.
+   *     - The +1 follows Asuncion et al. (2009), who recommend a +1 adjustment for EM.
+   */
+  def getTermSmoothing: Double = termSmoothing
+
+  def setTermSmoothing(termSmoothing: Double): this.type = {
+    require(termSmoothing > 0.0 || termSmoothing == -1.0,
+      s"LDA termSmoothing must be > 0 (or -1 for auto), but was set to $termSmoothing")
+    if (termSmoothing > 0.0 && termSmoothing <= 1.0) {
+      logWarning(s"LDA.termSmoothing was set to $termSmoothing, but for EM, we recommend > 1.0")
+    }
+    this.termSmoothing = termSmoothing
+    this
+  }
+
+  /**
+   * Maximum number of iterations for learning.
+   * (default = 20)
+   */
+  def getMaxIterations: Int = maxIterations
+
+  def setMaxIterations(maxIterations: Int): this.type = {
+    this.maxIterations = maxIterations
+    this
+  }
+
+  /** Random seed */
+  def getSeed: Long = seed
+
+  def setSeed(seed: Long): this.type = {
+    this.seed = seed
+    this
+  }
+
+  /**
+   * Learn an LDA model using the given dataset.
+   *
+   * @param documents  RDD of documents, where each document is represented as a vector of term
+   *                   counts plus an ID.  Document IDs must be >= 0.
+   * @return  Inferred LDA model
+   */
+  def run(documents: RDD[Document]): DistributedLDAModel = {
+    val topicSmoothing = if (this.topicSmoothing > 0) {
+      this.topicSmoothing
+    } else {
+      (50.0 / k) + 1.0
+    }
```

Comment:
Minor thing. How about moving the if-else to `getTermSmoothing`? For logic separation and also users can have a interface to collect the para